blockchain/src/blockchain/inherents.rs

```diff
@@ -55,6 +65,25 @@ impl Blockchain {
         inherents
     }
 
+    /// It creates a jail inherent from an equivocation proof. It expects a *verified* equivocation proof!
+    pub fn inherent_from_equivocation_proof(
+        &self,
+        block_number: u32,
+        equivocation_proof: &EquivocationProof,
+        txn_option: Option<&db::TransactionProxy>,
+    ) -> Inherent {
+        use self::EquivocationProof::*;
```

Comment:
Isn't it better to put it in the beginning of the file?

Reply:
Using it just here allows to use it for pattern matching. My thoughts was that you'd usually want to type the whole expression `EquivocationProof::Fork` instead of just `Fork`.

Reply:
Could've used the whole expressions here as well? IF that is what we would usually use then lets always use it?